app/controllers/planning_application/review_documents_controller.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class PlanningApplication
+  class ReviewDocumentsController < AuthenticationController
+    include CommitMatchable
+    include PlanningApplicationAssessable
+
+    before_action :set_planning_application
+    before_action :set_documents
+    before_action :ensure_planning_application_is_validated
+
+    def index
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def update_all
+      ActiveRecord::Base.transaction do
+        @planning_application.update!(review_documents_for_recommendation_status: status)
+
+        @documents.each do |document|
+          document.update!(
+            referenced_in_decision_notice: referenced_in_decision_notice?(document.id.to_s),
+            publishable: publishable?(document.id.to_s)
+          )
+        end
```

Comment:
Ideally would like one query here to do the multiple updates but can't use something like `update_all` as the value of the attributes (referenced_in_decision_notice / publishable) can be different for each document